File: src/musicgo/src/shared/classes/VoteViewModel.ts
import { UserViewModel } from './UserViewModel';
import { SongViewModel } from './SongViewModel';

export class VoteViewModel{
    public Positive: boolean = false;

    public User : UserViewModel | null = new UserViewModel();

    public TimeToPositive : number = 0;

    public Song: SongViewModel = new SongViewModel();
}

File: ViewModels/ViewModels/VoteViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ViewModels.ViewModels
{
    public class VoteViewModel : BaseViewModel
    {
        public bool Positive { get; set; }

        public SongViewModel Song { get; set; }

        public int TimeToPositive { get; set; }

        public UserViewModel User { get; set; }
    }
}
